### PR TITLE
Support Disabling Source Type Override Parameters

### PR DESCRIPTION
**Pull request by khola**
*Issue #, if available:*
https://github.com/aws-actions/aws-codebuild-run-build/issues/57

*Description of changes:*
Added additional input to allow for disabling the parameters `sourceTypeOverride` and `sourceLocationOverride`. Test cases added. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

**Comment by pavlovecjakePFG** [comment]
Thank you for working on this and submitting this pull request. We want this so we can use s3 as a source instead of GitHub. However, when testing your changes, s3 as a source fails with the following error:
"Source version should be empty for S3 folder source location". 
The sourceVersion also needs to be left unset in order to get s3 to work as a source.

File: action.yml [hunk 1]
```diff
@@ -13,6 +13,9 @@ inputs:
   env-vars-for-codebuild:
     description: 'Comma separated list of environment variables to send to CodeBuild'
     required: false
+  disable-source-override:
+    description: 'Set to `true` if you want do disable source repo override'
```
Comment (Mikequint94) [inline action.yml:17]: "if you want to" - typo

**Comment by mostmentor** [comment]
May I ask why this is not merged ? I need this feature very much in my project

**Comment by mostmentor** [comment]
@smailc I actually forked and tweaked it to do this https://github.com/mostmentor/aws-codebuild-run-build

File: code-build.js [hunk 2]
```diff
@@ -110,6 +110,8 @@ async function waitForBuildEndTime(sdk, { id, logs }, nextToken) {
 
 function githubInputs() {
   const projectName = core.getInput("project-name", { required: true });
+  const disableSourceOverride =
```
Comment (taoyong-ty) [inline code-build.js:113]: nit - For consistency, could you either update this new line to use core.getInput("disable-source-override", {required: false}) === "true", or remove all other { required: false }s from githubInputs()?

File: test/code-build-test.js [hunk 3]
```diff
@@ -94,6 +94,21 @@ describe("githubInputs", () => {
       .and.to.deep.equal(["one", "two", "three", "four"]);
   });
 
+  it("skips override when parameter is set to true", () => {
```
Comment (taoyong-ty) [inline test/code-build-test.js:97]: 1. could you also add a assertion in [build basic parameters for codeBuild.startBuild](https://github.com/aws-actions/aws-codebuild-run-build/blob/master/test/code-build-test.js#L60) that expect(test).to.haveOwnProperty("disableSourceOverride").and.to.equal(false);

2. the test case name is a bit confusing. should we just called it `"can process disable-source-override"`

File: test/code-build-test.js [hunk 4]
```diff
@@ -245,6 +260,17 @@ describe("inputs2Parameters", () => {
...
+
+  it("skips source override when disableSourceOverride is set to true", () => {
+    const test = inputs2Parameters({
+      projectName,
+      sourceVersion: sha,
+      owner: "owner",
+      repo: "repo",
+      disableSourceOverride: true,
+    });
+    expect(test).to.not.haveOwnProperty("sourceTypeOverride");
```
Comment (taoyong-ty) [inline test/code-build-test.js:272]: expect(test).to.not.haveOwnProperty("sourceLocationOverride");

**Review by taoyong-ty — COMMENTED** [review]
@khola  - Thank you so much for making this change. 

Apology for the delayed review. If you are still interested in wrapping up and merging this pull request, please address the comment and resolve the conflicts. 

I am ready to approve and release once the comments have been addressed

**Comment by taoyong-ty** [comment]
close this in favor of https://github.com/aws-actions/aws-codebuild-run-build/pull/102

**Pull request closed without merging**